Kconfig

```diff
@@ -0,0 +1,126 @@
+mainmenu "Crazyflie Platform Configuration"
+
+menu "Build and compiler options"
+
+config CROSS_COMPILE
+	string "Cross-compiler tool prefix"
+    default "arm-none-eabi-"
+	help
+	  Same as running 'make CROSS_COMPILE=prefix-' but stored for
+	  default make runs in this application build directory.  You don't
+	  need to set this unless you want the configured application build
+	  directory to select the cross-compiler automatically.
+
+config DEBUG
+    bool "Enable debug build"
+    default n
+    help
+        Enable debug symbols and extra output from firnware.
+
+endmenu
+
+menu "Platform configuration"
+
+choice
+	prompt "Platform to build"
+	default CONFIG_PLATFORM_CF2
+
+config PLATFORM_CF2
+    bool "Build for CF2"
+    select SENSORS_BMI088_BMP388
+    select SENSORS_MPU9250_LPS25H
+
+config PLATFORM_BOLT
+    bool "Build for Bolt"
+    select SENSORS_BMI088_BMP388
+    select SENSORS_BMI088_SPI
+
+config PLATFORM_TAG
+    bool "Build for the roadrunner"
+    select SENSORS_BMI088_BMP388
+
+endchoice
+
+endmenu
+
+menu "Sensor configuration"
+
+config SENSORS_MPU9250_LPS25H
+    bool "Support for mpu9250 and lps25h sensors"
+    default n
+    help
+    Include support for the Bosch bmi088 inertial and bmp388
+    barometric sensors
+
+config SENSORS_BMI088_BMP388
+    bool "Support for bmi088 and bmp388 sensors"
+    default n
+    select SENSORS_BMI088_I2C
+    help
+    Include support for the Bosch bmi088 inertial and bmp388
+    barometric sensors
+
+config SENSORS_BOSCH
+    bool "Include Bosch sensors"
+    default n
+
+config SENSORS_IGNORE_BAROMETER_FAIL
+    bool "Ignore failure from barometer"
+    depends on SENSORS_BMI088_BMP388
+    default n
+
+config SENSORS_BMI088_SPI
+    bool "Support for SPI communincation with the bmi088 sensor"
+    depends on SENSORS_BMI088_BMP388
+    default n
+    help
+        Include support using SPI with the Bosch bmi088 inertial sensor
+
+config SENSORS_BMI088_I2C
+    bool "Support for I2C communincation with the bmi088 sensor"
+    depends on SENSORS_BMI088_BMP388
+    default n
+    help
+        Include support using I2C with the Bosch bmi088 inertial sensor
+
+endmenu
+
+menu "App layer configuration"
+
+source "app_api/Kconfig"
+
+config APP_ENABLE
+    bool "Enable app entry point"
+    default n
+    help
+      When enabled the firmware will call a function void appMain() from a
+      task after the startup sequence has completed. This function should not
+      return.
+
+config APP_STACKSIZE
+    int "Set the task stack size"
+    depends on APP_ENABLE
+    default 300
+    help
+      Size is in 32bit word (4 Bytes). The default is 300 (1.2KBytes)
+
+config APP_PRIORITY
+    int "Set the task priority"
+    depends on APP_ENABLE
+    default 0
+    help
+      Set the task priority between 0 and 5. Default is 0 (same as IDLE).
+
+endmenu
+
+menu "Expansion deck configuration"
+
+config DECK_FORCE
+    string "Force loading of a deck driver"
```

Comment:
The string now looks like '() Force loading of a deck driver' but it it unclear that that line does any thing specific or goes to a different windo. Perhaps better to do:
'Force loading of a manual deck driver --->'

(also indicating that it is not a generic one is important)

Reply:
> The string now looks like '() Force loading of a deck driver' but it it unclear that that line does any thing specific or goes to a different windo. Perhaps better to do: 'Force loading of a manual deck driver --->'
> 
> (also indicating that it is not a generic one is important)

Hmm, ok, is this better?

![force](https://user-images.githubusercontent.com/974401/152958037-03fb78b7-8895-4ad6-a921-acd2f1bc3721.png)

Reply:
Perhaps better to add the arrow to make it consistent with the main menu item ? That indicates that there is a next dialog upcoming

Reply:
> Perhaps better to add the arrow to make it consistent with the main menu item ? That indicates that there is a next dialog upcoming

Yeah, I am not sure about this, since we are not doing anything special in term of Kconfig, this is how it looks when you specify the type string for a configuration option. Perhaps it should conform to how other Kconfig configs look. We do not draw the arrows on menu items, the Kbuild system does.

Busybox has strings they look like:

![busybox](https://user-images.githubusercontent.com/974401/152971149-fc024b67-bb32-4318-8f2f-20c95e8264d2.png)

![udhcp](https://user-images.githubusercontent.com/974401/152971164-8ca00617-d311-4483-b1a8-f5b828683349.png)

Perhaps its more the fact that we have no default? Should we have one? So it would look more natural?
Like so:
![default](https://user-images.githubusercontent.com/974401/152971909-8853b048-1c4d-4d78-8bbe-d46c8e86114a.png)


With help:
![help-none](https://user-images.githubusercontent.com/974401/152971915-6d439c60-7c04-49b4-a5a5-2e25053ee15a.png)

Reply:
ah oke, yeah then it's a bit more clear of what it supposed to be